### PR TITLE
Cannabis extract overhaul

### LINKED ISSUE #52743: "high quality distillate (heads)" recipe byproduct is excessive/inadequate
### Describe the bug

the recipe for "high quality distillate (heads)" gives back as byproduct the same material it needs to be crafted, in increased quantity... literally creating items from nothing

### Steps To Reproduce

1 spawn materials needed to make the whole ordeal
2 craft it
3 observe your starting ingredient "high quality distillate (tails)" growing in size

### Expected behavior

i'm not an expert of oil refining, but this recipe goes against the laws of physics creating items from nothing

### Screenshots

image

### Versions and configuration

windows 10
version 6db9695
build 2021-11-10-0755

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by onura46**
#### Summary
Balance "Add medical CBD extracts, rebalance cannabis extraction recipes and related drop rates"

#### Purpose of change
Adds medical-grade CBD extracts, disambiguates between extant THC extracts, modifies cannabis extraction recipes to be more consistent and realistic, and modifies drops for related items.

#### Describe the solution
**TL;DR**: This is an overhaul of the cannabis extraction system and many of its related items aimed at improving realism (somewhat) and consistency of some recipes. Please check "Additional Context" for research related to this PR and a breakdown of the extract crafting system.

**"Medical-grade CBD extracts" have been added as an alternative to THC extracts.** They are ~30% more potent as painkillers than their THC variants and do not cause you to get high, but also require the highest-end, closed-loop extraction equipment to produce (this was a gameplay-based decision to urge you to plant cannabis and seek out closed-loop extraction methods for renewable painkillers; realism sticklers, please be kind). They are not made from a different variety of cannabis plant, I'm sorry. (Realism sticklers... try not to think about it.) This could be changed with different varieties of cannabis in the future, but that was a little out of the scope of this PR.

**All extract recipes have been changed for improved realism.** Solvent usage has been changed to use more in open blasting recipes. Non-solvent-based recipes (rosin presses mainly, I'm thinking) are not in-game at the moment. Tool recipes have been changed to be slightly more realistic. I left the recipes for vacuum pumps and ovens. Even though they are mechanically pretty complex, better-than-improvised DIY solutions have been demonstrated online. Please check "Additional Context" for more information on this. 

**Filter paper is no longer craftable or deconstructable.** Real-world filter paper is created in a factory process by treating not only paper, but fiberglass, plastic, nanocarbons, etc. with special chemicals to specifically define the width of its pores. No survivor should be able to craft it.

**Item groups have been slightly changed.** Plastic gaskets are now in "supplies_hardware" and filter paper is now a part of "tools_science".

_NOTE: This PR still currently suffers from a bug (?) pointed out in #59230 and ethanol-based recipes currently return an entire stack (250) of ethanol per unit rather than 1 unit like they should. I'll edit this section of the code to fix this if required in the future._

This PR should also close #52743.

#### Describe alternatives you've considered
I'm torn between expanding the system and simplifying the process further by simply making one type of each: "CBD extract" and "THC extract". However, I hope the research here proves that there's enough depth to justify having slightly more options for a needy survivor. In real chemistry, the tools to extract cannabis oils are also used to extract many other essential oils, opening up another avenue for crafting advanced herbal medicines.

Certain item recipes could be changed individually, but this PR addresses numerous issues brought up elsewhere, notably #52743. Perhaps this PR doesn't even go far enough disambiguating THC and CBD cannabis plants for crafting purposes. This could be expanded upon in the future.

Currently there's also no way to collect used butane solvent back into a container from a closed-loop system, since butane is not yet a material in-game. This should probably be changed in the future, and the current way around this is to assume that the butane is trapped within the closed-loop system itself and recycled every time.

#### Testing
Tested on my own compiled version, confirming recipes appear as expected.

#### Additional context
Cannabis extraction in C:DDA can currently be performed with two solvents:
- Butane: Boils off at room temperature over time or over low heat; can be used for cheaper, improvised extraction methods.
- Ethanol: Requires heat to boil off; can be used for the largest, most efficient extraction methods.

Cannabis can currently be extracted into concentrates in C:DDA in two ways:
- Open-loop extraction (open blasting): involves submerging cannabis in a pressurized container (a pipe with a clamp, in this PR) with butane solvent for a time. Solvent boils off into the air (which has the potential to cause explosions IRL and could in-game on critical failures in the future).
- Closed-loop extraction: involves building or finding a closed-loop extraction system (small or large) and using either butane or ethanol as a solvent to extract the cannabinoids. If you use ethanol, this is used in the recipe then returned to you. If you use butane, it's simply a part of the closed-loop system itself and assumed to be recycled every time.

Cannabis extract is currently refined in C:DDA in two ways, by boiling off and by vacuum oven (this isn't super obvious, but is implied by which recipe and solvent you choose to use)
- Boil off (butane): takes place at very low temperature (~90-100 degrees F) and normal atmospheric pressure over time, and results in less product in-game.
- Boil off (ethanol): requires heat, is totally recycled as a byproduct in closed loop extractions, and can be used either on its own or in addition to a vacuum oven for further refining.
- Vacuum oven: A special tool that lowers pressure inside it and applies low heat over a very long time to further refine concentrates into shatter. It requires a bowl made of aluminum foil (irl parchment paper works better). Closed-loop extraction methods using a vacuum oven result in the most product. Using a vacuum oven in Cataclysm is honestly kind of a flex because shatter recipes take 3 or 4 days to complete.

Therefore:
- Ethanol solvent + Large closed-loop system + Vacuum Oven = **Best Method**
- Butane solvent + No closed-loop system + No vacuum pump or oven = **Worst Method**

_This system was created by @NastyNate2612 and further improved by @AtomicFox556, and I wanted to collect as much research as I could for anyone who might want to expand on this in the future:_

[Making BHO with an open-loop extraction method Part 1 (open blasting) (Video)](https://youtu.be/tn6TJcpiq9c)
[Making BHO with an open-loop extraction method Part 2 (vacuum pump + glass flask equivalent) (Video)](https://youtu.be/U0Pr9YI33f0)
[Making BHO in a small closed-loop extraction system with butane (Video)](https://youtu.be/GGyeH8UA5jI)
[Homemade vacuum pump recipe (Video)](https://youtu.be/BeiW_37rS-U)
[Purging both butane and ethanol extracts in a vacuum oven (Video)](https://youtu.be/DbXPna0PjHk)
[More info on vacuum ovens (Article)](https://blog.lunatechequipment.com/bho-201)

### CODE REVIEW

**Comment by onura46** [comment]
This PR likely closes #52743 (which was already partially solved) and also affects one recipe in Innawood!

**Review by Light-Wave — APPROVED** [review]

**Comment by AtomicFox556** [comment]
> Currently there's also no way to collect used butane solvent back into a container from a closed-loop system, since butane is not yet a material in-game. This should probably be changed in the future, and it means you don't get butane on your recipes like you can ethanol.

That's why I've originally made the "butane container" tools a requirement for crafting closed-loop extractors in #56703, assuming that the solvent is stored somewhere within the item. It only makes sense if you could recover *all* the solvent, which I was thinking is possible. If it's not, could add some solvent as a component requirement to extraction recipes themselves, to replace the consumed part of solvent.

**Comment by onura46** [comment]
@AtomicFox556 Are there any other changes you would suggest making here (besides the extant stuff like the material issue) before I call this ready to merge? Do any recipes seem off? I think the extraction process is a little complex, so I appreciate feedback from anyone who understands what's going on. Thanks <3

**Comment by AtomicFox556** [comment]
> @AtomicFox556 Are there any other changes you would suggest making here (besides the extant stuff like the material issue) before I call this ready to merge? Do any recipes seem off? I think the extraction process is a little complex, so I appreciate feedback from anyone who understands what's going on. Thanks <3

I'd change extraction recipes making use of closed-loop systems to not consume butane by just not including butane in recipe, with the assumption being that butane is trapped within the system. Even if it's not how it's *usually* done IRL, it could be done like that and appears to be the most convenient way to do it as gas storage is a bit awkward at the moment.

**Comment by onura46** [comment]
I'm gonna rework this a little, actually, because it wasn't the most well-conceived PR. It could probably be split up as well, because there are typo fixes, item group changes, and (definitely unnecessary) recipe additions all in one.
